src/olympia/translations/tests/test_models.py

```diff
@@ -701,6 +702,57 @@ def test_external_text_link(self, get_outgoing_url_mock):
         assert doc('b')[0].text == 'markup'
 
 
+class PurifiedMarkdownTranslationTest(TestCase):
+    def test_output(self):
+        assert isinstance(PurifiedMarkdownTranslation().__html__(), str)
+
+    def test_raw_text(self):
+        s = '   This is some text   '
+        x = PurifiedMarkdownTranslation(localized_string=s)
+        assert x.__html__() == 'This is some text'
+
+    def test_markdown(self):
+        s = '__bold text__ or _italics_<b>not bold</b>'
+        x = PurifiedMarkdownTranslation(localized_string=s)
+        assert x.__html__() == (
+            '<strong>bold text</strong> or <em>italics</em>'
+            '&lt;b&gt;not bold&lt;/b&gt;'
+        )
+
+    def test_html(self):
+        s = '<script>some naughty xss</script>'
+        x = PurifiedMarkdownTranslation(localized_string=s)
+        assert x.__html__() == '&lt;script&gt;some naughty xss&lt;/script&gt;'
+
+    def test_internal_link(self):
+        s = '**markup** [bar](http://addons.mozilla.org/foo)'
+        x = PurifiedMarkdownTranslation(localized_string=s)
+        doc = pq(x.__html__())
+        links = doc('a[href="http://addons.mozilla.org/foo"][rel="nofollow"]')
+        assert links[0].text == 'bar'
+        assert doc('strong')[0].text == 'markup'
+
+    @patch('olympia.amo.urlresolvers.get_outgoing_url')
+    def test_external_link(self, get_outgoing_url_mock):
+        get_outgoing_url_mock.return_value = 'http://external.url'
+        s = '**markup** [bar](http://example.com)'
+        x = PurifiedMarkdownTranslation(localized_string=s)
+        doc = pq(x.__html__())
+        links = doc('a[href="http://external.url"][rel="nofollow"]')
+        assert links[0].text == 'bar'
+        assert doc('strong')[0].text == 'markup'
+
+    @patch('olympia.amo.urlresolvers.get_outgoing_url')
+    def test_external_text_link(self, get_outgoing_url_mock):
+        get_outgoing_url_mock.return_value = 'http://external.url'
+        s = '**markup** http://example.com'
+        x = PurifiedMarkdownTranslation(localized_string=s)
+        doc = pq(x.__html__())
+        links = doc('a[href="http://external.url"][rel="nofollow"]')
+        assert links[0].text == 'http://example.com'
+        assert doc('strong')[0].text == 'markup'
+
```

Comment:
Can you add a test of mixed html and markdown, simulating someone editing an existing html string and adding markdown to it. We would expect the html to be stringified and the markdown to be formatted to html right?

Reply:
`test_markdown` does have a case of mixed HTML and markdown, and yeah -- the HTML is stringified, the markdown is formatted.

Reply:
ah I missed that in the name `test_mixed_markdown_html` maybe?